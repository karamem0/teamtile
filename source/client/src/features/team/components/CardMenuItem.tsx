//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { EventHandler } from '../../../types/Event';
import Presenter from './CardMenuItem.presenter';

interface CardMenuItemProps {
  icon?: React.ReactElement,
  title?: string,
  onClick?: EventHandler
}

function CardMenuItem(props: Readonly<React.PropsWithChildren<CardMenuItemProps>>) {

  const {
    icon,
    title,
    onClick
  } = props;

  return (
    <Presenter
      icon={icon}
      title={title}
      onClick={onClick} />
  );

}

export default CardMenuItem;
