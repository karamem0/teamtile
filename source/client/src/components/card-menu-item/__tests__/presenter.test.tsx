//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import CardMenuItem from '../presenter';

beforeEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
});

describe('ChannelMenuItem', () => {

  it('create shapshot when menu item is not open', async () => {
    const params = {
      icon: <div data-testid="Icon" />,
      content: <div data-testid="Content" />
    };
    render(<CardMenuItem {...params} />);
    expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
  });

});
