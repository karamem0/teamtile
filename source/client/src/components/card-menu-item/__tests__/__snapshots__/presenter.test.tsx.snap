// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelMenuItem create shapshot when menu item is not open 1`] = `
.emotion-0 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto auto;
  grid-gap: 0.25rem;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  cursor: pointer;
}

.emotion-1 {
  width: 1rem;
  height: 1rem;
  line-height: 1rem;
}

.emotion-2 {
  line-height: 0.75rem;
}

<body>
  <div>
    <span
      class="ui-text emotion-0"
      data-uses-unhanded-props="true"
      role="button"
    >
      <span
        class="emotion-1"
      >
        <div
          data-testid="Icon"
        />
      </span>
      <span
        class="ui-text emotion-2"
        data-uses-unhanded-props="true"
      >
        <div
          data-testid="Content"
        />
      </span>
    </span>
  </div>
</body>
`;
